src/main/java/fr/openent/competences/controllers/CompetenceNoteController.java

```diff
@@ -462,34 +440,37 @@ public void deleteCompetencesNotesDevoir(final HttpServerRequest request) {
         competencesNotesService.dropCompetencesNotesDevoir(oIdsJsonArray, arrayResponseHandler(request));
     }
 
-    private void callGetCompetenceNote(String idClasse, Long idPeriode, Integer typeClasse, List<String> idDomaines,
+    private void callGetCompetenceNote(String idClasse, Long idPeriode, Integer typeClasse, String idStructure,
                                        final HttpServerRequest request) {
-        new DefaultUtilsService(this.eb).studentIdAvailableForPeriode(idClasse, idPeriode, typeClasse,
-                new Handler<Either<String, JsonArray>>() {
-                    @Override
-                    public void handle(Either<String, JsonArray> event) {
-                        if (event.isRight()) {
-                            JsonArray queryResult = event.right().getValue();
-                            List<String> idEleves = new ArrayList<String>();
-
-                            if (queryResult != null) {
-                                for (int i = 0; i < queryResult.size(); i++) {
-                                    idEleves.add(queryResult.getString(i));
-                                }
-                            }
-                            if (idDomaines != null) {
-                                callCompetencesNotesDomaineService(idEleves, idPeriode, idDomaines, request);
-                            } else {
-                                callCompetencesNotesService(idEleves, idPeriode, request);
-                            }
+        new DefaultUtilsService(this.eb).studentIdAvailableForPeriode(idClasse, idPeriode, typeClasse, event -> {
+            if (event.isRight()) {
+                JsonArray queryResult = event.right().getValue();
+                List<String> idEleves = new ArrayList<String>();
+
+                if (queryResult != null) {
+                    for (int i = 0; i < queryResult.size(); i++) {
+                        idEleves.add(queryResult.getString(i));
+                    }
+                }
+                structureOptionsService.getIsAverageSkills(idStructure, responseOption -> {
 
-                        } else {
-                            Renders.notFound(request,
-                                    "Error :can not get CompNotes of groupe : " + idClasse);
-                            log.error("Error :can not get compNotes of groupe : " + idClasse);
-                        }
+                    if(responseOption.isLeft()) {
+                        Renders.notFound(request,"Impossible de recuperer le type de calcul des competences notes \n"
```

Comment:
Attention tu envoies une erreur "not found" mais comme tu return pas, il continuera le traitement et tu te risques à une erreur de http